test/inductor/test_cpu_repro.py

```diff
@@ -1376,6 +1376,53 @@ def fn(a):
                         if simdlen != 1:
                             assert metrics.generated_cpp_vec_kernel_count == 2
 
+    def test_horizontal_fusion(self):
+        def fn(a, b, c, idx):
+            _a = torch.index_select(a, dim=0, index=idx)
+            _b = torch.index_select(b, dim=0, index=idx)
+            _c = torch.index_select(c, dim=0, index=idx)
+            return _a, _b, _c
+
+        with config.patch({"cpp.max_horizontal_fusion_size": 0}):
+            metrics.reset()
+            a = torch.randn(size=(4, 16), dtype=torch.bfloat16)
+            b = torch.randn(size=(4, 16), dtype=torch.bfloat16)
+            c = torch.randn(size=(4, 16), dtype=torch.bfloat16)
+            idx = torch.zeros(size=[4], dtype=torch.int64)
+            opt_fn = torch._dynamo.optimize("inductor")(fn)
+            opt_fn(a, b, c, idx)
+            self.assertTrue(same(fn(a, b, c, idx), opt_fn(a, b, c, idx)))
```

Comment:
Can you also check the number of generated kernels here to make sure the fusion size takes effect?

Reply:
Actually, the codegen for this eager mode still produces a single kernel.

Reply:
Do you mean both fused and non-fused cases generate a single kernel? Or it is a single kernel group but multiple kernels if it is not fused?

Reply:
```c++
extern "C" void kernel(const long* in_ptr0,
                       const bfloat16* in_ptr1,
                       const bfloat16* in_ptr2,
                       const bfloat16* in_ptr3,
                       bfloat16* out_ptr0,
                       bfloat16* out_ptr1,
                       bfloat16* out_ptr2)
{
    #pragma omp parallel num_threads(48)
    {
        {
            #pragma omp for
            for(long i0=static_cast<long>(0L); i0<static_cast<long>(4L); i0+=static_cast<long>(1L))
            {
                #pragma GCC ivdep
                for(long i1=static_cast<long>(0L); i1<static_cast<long>(32L); i1+=static_cast<long>(1L))
                {
                    auto tmp0 = in_ptr0[static_cast<long>(i0)];
                    auto tmp1 = in_ptr1[static_cast<long>(i1 + (32L*tmp0))];
                    out_ptr0[static_cast<long>(i1 + (32L*i0))] = tmp1;
                }
            }
        }
        {
            #pragma omp for
            for(long i0=static_cast<long>(0L); i0<static_cast<long>(4L); i0+=static_cast<long>(1L))
            {
                #pragma GCC ivdep
                for(long i1=static_cast<long>(0L); i1<static_cast<long>(32L); i1+=static_cast<long>(1L))
                {
                    auto tmp0 = in_ptr0[static_cast<long>(i0)];
                    auto tmp1 = in_ptr2[static_cast<long>(i1 + (32L*tmp0))];
                    out_ptr1[static_cast<long>(i1 + (32L*i0))] = tmp1;
                }
            }
        }
        {
            #pragma omp for
            for(long i0=static_cast<long>(0L); i0<static_cast<long>(4L); i0+=static_cast<long>(1L))
            {
                #pragma GCC ivdep
                for(long i1=static_cast<long>(0L); i1<static_cast<long>(32L); i1+=static_cast<long>(1L))
                {
                    auto tmp0 = in_ptr0[static_cast<long>(i0)];
                    auto tmp1 = in_ptr3[static_cast<long>(i1 + (32L*tmp0))];
                    out_ptr2[static_cast<long>(i1 + (32L*i0))] = tmp1;
                }
            }
        }
    }
}
```

Reply:
Oh, captured your point. Fixed it.